File: 7_Function_Overriding.py

# Runtime Polymorphism -> Dynamic Binding | Overriding | Late Binding

# CompileTime Polymorphism-> Static Binding | Overloading | Early Binding


class Vehicle:
    def string_engine(self):
        print("Staring Vehicle")

class Car (Vehicle):
    def string_engine(self):
        print("Staring Car Engine")

class Bike (Vehicle):
    def string_engine(self):
        print("Staring Bike Engine")


car = Car()

car.string_engine() # Output: Starting Car Engine

bike = Bike()

bike.string_engine() # Output: Starting Bike Engine

vehicle = Vehicle()

vehicle.string_engine() # Output: Starting Vehicle

File: 7_Function_Overriding.cpp


#include <bits/stdc++.h>
using namespace std;

class Vehicle
{
public:
    virtual void display() { cout << "This is a vehicle\n"; }
};

class Car : public Vehicle{
    public:
    void display() { cout << "This is a car\n"; }
};

class Bike : public Vehicle{public:
    void display() { cout << "This is a bike\n"; }
};


int
main()
{
    Vehicle * v = new Vehicle;
    v->display();
    
    Car *c = new Car;
    c->display();

    Bike *b=new Bike;
    b->display();
    

    delete v;
    delete c;
    delete b;
    
    return 0;
}